numba/cpython/randomimpl.py

```diff
@@ -449,93 +447,106 @@ def uniform_impl(context, builder, sig, args):
     res = uniform_impl(context, builder, sig, args, "np")
     return impl_ret_untracked(context, builder, sig.return_type, res)
 
+
 def uniform_impl(context, builder, sig, args, state):
     state_ptr = get_state_ptr(context, builder, state)
     a, b = args
     width = builder.fsub(b, a)
     r = get_next_double(context, builder, state_ptr)
     return builder.fadd(a, builder.fmul(width, r))
 
-@glue_lowering("random.triangular", types.Float, types.Float)
-def triangular_impl_2(context, builder, sig, args):
-    fltty = sig.return_type
-    low, high = args
-    state_ptr = get_state_ptr(context, builder, "py")
-    randval = get_next_double(context, builder, state_ptr)
 
-    def triangular_impl_2(randval, low, high):
-        u = randval
+@overload(random.triangular)
+def triangular_impl_2(low, high):
+    def _impl(low, high):
+        u = random.random()
         c = 0.5
         if u > c:
             u = 1.0 - u
             low, high = high, low
         return low + (high - low) * math.sqrt(u * c)
 
-    res = context.compile_internal(builder, triangular_impl_2,
-                                    signature(*(fltty,) * 4),
-                                    (randval, low, high))
-    return impl_ret_untracked(context, builder, sig.return_type, res)
+    if isinstance(low, types.Number) and isinstance(high, types.Number):
+        return _impl
+
+
+@overload(random.triangular)
+def triangular_impl_3(low, high, mode):
+    if (isinstance(low, types.Number) and isinstance(high, types.Number) and
+       isinstance(mode, types.Number)):
+        def _impl(low, high, mode):
+            if high == low:
+                return low
+            u = random.random()
+            c = (mode - low) / (high - low)
+            if u > c:
+                u = 1.0 - u
+                c = 1.0 - c
+                low, high = high, low
+            return low + (high - low) * math.sqrt(u * c)
+
+        return _impl
+
+
+@overload(np.random.triangular)
+def triangular_impl_3(low, mode, high):
+    if (isinstance(low, types.Number) and isinstance(mode, types.Number) and
+            isinstance(high, types.Number)):
+        def _impl(low, mode, high):
+            if high == low:
+                return low
+            u = np.random.random()
+            c = (mode - low) / (high - low)
+            if u > c:
+                u = 1.0 - u
+                c = 1.0 - c
+                low, high = high, low
+            return low + (high - low) * math.sqrt(u * c)
+
+        return _impl
+
+
+@overload(np.random.triangular)
+def triangular_impl(low, high, mode, size):
+    if isinstance(size, types.NoneType):
+        return lambda low, high, mode, _size: np.random.triangular(low, high,
```

Comment:
```suggestion
        return lambda low, high, mode, size: np.random.triangular(low, high,
```

`@overload` requires the parameters to match exactly for the returned impl and the decorated func; otherwise you get error like:

```pytb
 Rejected as the implementation raised a specific error:
     InternalError: Typing and implementation arguments differ in argument names.
   Typing signature:         (low, high, mode, size)
   Implementation signature: (low, high, mode, _size)
   Difference: {<Parameter "_size">, <Parameter "size">}
```

Reply:
Also, the `func(..., size=None)` path is not exercised by the tests.